R/utils.R

```diff
@@ -66,21 +70,40 @@ parse_answer = function(api_answer, multiple = FALSE) {
     journals = xml_text(xml_find_all(xml_source, "//jtitle"))
     issns = xml_text(xml_find_all(xml_source, "//issn"))
 
+    journal_df = data.frame(title = journals,
+                            issn  = issns)
+
     if (!multiple) {
       warning("Select one journal from the provided list or enable multiple = ",
               "TRUE")
 
-      return(data.frame("title" = journals,
-                        "issn" = issns))
+      return(journal_df)
     } else {
 
       message("Recursively fetching data from each journal. ",
               "This may take some time...")
 
-      # Dropping silently missing ISSNs
-      issns = issns[lapply(issns, nchar) > 0]
-
-      return(do.call(rbind.data.frame, lapply(issns, rr_journal_issn)))
+      # Retrieve RoMEO data for all matched journals
+      # if the ISSN isn't available try to match using title otherwise use ISSN
+      result_df = apply(journal_df, 1, function(x) {
+        if (x["issn"] != "") {
+          journal_policy = rr_journal_issn(x["issn"], check_key(key))
+
+        } else {
+          journal_policy = tryCatch({
+            rr_journal_name(x["title"], check_key(key), qtype = "exact")
+          },
+          error = function(err) {
+            return(data.frame(title = x["title"],
+                              issn = x["issn"],
+                              preprint    = NA,
+                              postprint   = NA,
+                              pdf         = NA,
+                              romeocolour = NA))
```

Comment:
Given the current master, this should also include the embargo policies before we merge it